core/lib/zksync_core/src/tee_verifier_input_producer/mod.rs

```diff
@@ -0,0 +1,262 @@
+            .block_on(connection_pool.connection())
+            .context("failed to get connection for TeeVerifierInputProducer")?;
+
+        let miniblocks_execution_data = rt_handle.block_on(
+            connection
+                .transactions_dal()
+                .get_l2_blocks_to_execute_for_l1_batch(l1_batch_number),
+        )?;
+
+        let last_batch_miniblock_number = miniblocks_execution_data.first().unwrap().number - 1;
+
+        let l1_batch_header = rt_handle
+            .block_on(connection.blocks_dal().get_l1_batch_header(l1_batch_number))
+            .with_context(|| format!("header is missing for L1 batch #{l1_batch_number}"))?
+            .unwrap();
+
+        let l1_batch_params_provider = rt_handle
+            .block_on(L1BatchParamsProvider::new(&mut connection))
+            .context("failed initializing L1 batch params provider")?;
+
+        let first_miniblock_in_batch = rt_handle
+            .block_on(
+                l1_batch_params_provider
+                    .load_first_l2_block_in_batch(&mut connection, l1_batch_number),
+            )
+            .with_context(|| {
+                format!("failed loading first miniblock in L1 batch #{l1_batch_number}")
+            })?
+            .with_context(|| format!("no miniblocks persisted for L1 batch #{l1_batch_number}"))?;
+
+        // In the state keeper, this value is used to reject execution.
+        // All batches have already been executed by State Keeper.
+        // This means we don't want to reject any execution, therefore we're using MAX as an allow all.
+        let validation_computational_gas_limit = u32::MAX;
+
+        let (system_env, l1_batch_env) = rt_handle
+            .block_on(l1_batch_params_provider.load_l1_batch_params(
+                &mut connection,
+                &first_miniblock_in_batch,
+                validation_computational_gas_limit,
+                l2_chain_id,
+            ))
+            .context("expected miniblock to be executed and sealed")?;
+
+        let pg_storage = PostgresStorage::new(
+            rt_handle.clone(),
+            connection,
+            last_batch_miniblock_number,
+            true,
+        );
+        let mut real_storage_view = StorageView::new(pg_storage);
+
+        let used_contracts = l1_batch_header
+            .used_contract_hashes
+            .into_iter()
+            .filter_map(|hash| {
+                ReadStorage::load_factory_dep(&mut real_storage_view, u256_to_h256(hash))
+                    .map(|bytes| (u256_to_h256(hash), bytes))
+            })
+            .collect();
+
+        info!("Started execution of l1_batch: {l1_batch_number:?}");
+
+        let tee_verifier_input = TeeVerifierInput::new(
+            prepare_basic_circuits_job,
+            miniblocks_execution_data,
+            l1_batch_env,
+            system_env,
+            used_contracts,
+        );
+
+        // TODO: remove these 2 lines after successful testnet runs
```

Comment:
If this TODO is going to remain after merge, please add a linear task number to it (e.g. `TODO (xxx-yyy): Remove once tested on testnet`), so that we don't loose track of it.

Reply:
added
https://linear.app/matterlabs/issue/SEC-263/zksync-era-remove-proof-verifying-in-input-producer